main.rb

```diff
@@ -0,0 +1,95 @@
+module Enumerable
+
+    def my_each
+    unless !block_given?
+      for item in self
+        yield(item)
+      end
+    end
+      return self
+    end
+  
+    def my_each_with_index
+    unless !block_given?
+      for item_index in (0...self.size)
+        yield(self[item_index], item_index)
+      end
+    end
+      return self
+    end
```

Comment:
Remove return, is called an explicit return. And it is a bad practice. Just `self` is enough to handle the case.
`my_each_with_index` identical and returns the same thing as ruby's [`each_with_index`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-each_with_index). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_each_with_index.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- Should return an enumerator if no block given.